File: static/CWE-89/csharp/vulnerable.cs
// CWE-89: Improper Neutralization of Special Elements used in an SQL Command ('SQL Injection')
// User input is concatenated directly into a SQL query string and executed.
// Vulnerable sink: new SqlCommand(query, conn) where query = "... " + userInput

using System.Data.SqlClient;

namespace GuineaPig.Sast
{
    public class CWE_89
    {
        public void LookupUser(string userInput)
        {
            using (var conn = new SqlConnection("Server=db;Database=app;Integrated Security=true;"))
            {
                conn.Open();
                string query = "SELECT * FROM Users WHERE Name = '" + userInput + "'";
                using (var cmd = new SqlCommand(query, conn))
                using (var reader = cmd.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        System.Console.WriteLine(reader[0]);
                    }
                }
            }
        }
    }
}

File: static/CWE-89/javascript/vulnerable.js
// CWE-89: SQL Injection
// Untrusted request input is concatenated directly into a SQL query string.
// Vulnerable sink: connection.query(`... ${req.params.id}`)

const express = require('express');
const mysql = require('mysql');

const app = express();
const connection = mysql.createConnection({
  host: 'localhost',
  user: 'root',
  password: 'root',
  database: 'app',
});

app.get('/users/:id', (req, res) => {
  const userId = req.params.id;
  const query = `SELECT * FROM users WHERE id = ${userId}`;
  connection.query(query, (err, results) => {
    if (err) return res.status(500).send('error');
    res.json(results);
  });
});

app.listen(3000);
